### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by gerasiov**
Separate platform-specific code from mqtt client.
Simplify logic a little bit.

### CODE REVIEW

**Comment by gerasiov** [comment]
I'm going to implement sound control in lumimqtt
Prior this work I'm refactoring the code a little bit.
Could be split in separate PR's if you'd like it more.

**Comment by devbis** [comment]
Thank you for your PR! The code looks very promising and really simplifies it in some places.
But there are some points that I strongly disagree with.
1. Changing line length to 100 chars leads to wrapping lines when viewing diffs on the Github

2. Implementing sound is not a priority as there is no good solution for auto-discover audio service in Home Assistant. Though it has a native and more convenient way to use TTS and play music with MPD.

I was thinking of it as a separate service for entities without autodiscovery for non-HA users.

Anyway, the PR looks good, with the line length exception.

**Comment by gerasiov** [comment]
>Changing line length to 100 chars leads to wrapping lines when viewing diffs on the Github
Ok, I'll remove this commit from the branch and reformat long lines.

>Implementing sound is not a priority as there is no good solution for auto-discover audio service in Home Assistant. Though it has a native and more convenient way to use TTS and play music with MPD.
Well, I need simple solution for my HA-based (but not lumi-hosted) automation to implement TTS messages and music files playback (e.g. doorbell) So I'd like to implement some API similar to https://github.com/Beetle-II/lumi but with python, not node.js =)

PS I'm new to HA/mqtt world and could do somethings not optimal here, but would like to have clear solution for the task I mentioned.

File: lumimqtt/sensors.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            try:
+                with open('/sys/class/gpio/export', 'w') as f:
+                    f.write(str(gpio))
+                with open(f'/sys/class/gpio/gpio{gpio}/direction', 'w') as f:
+                    f.write('in')
+            except OSError as err:
+                logger.error(f"Can not setup {name} sensor: {err}")
+
+    def get_value(self):
+        return 'ON' if self.read_raw() else 'OFF'
```
Comment (gerasiov) [inline lumimqtt/sensors.py:42]: Lost int() here. Or may be more strict check.

File: lumimqtt/sensors.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        'device_class': 'illuminance',
+        'unit_of_measurement': 'lx',
+    }
+
+    def get_value(self):
+        raw_value = self.read_raw()
+        try:
+            return int(int(raw_value) * self.COEFFICIENT)
+        except ValueError:
+            return raw_value
```
Comment (gerasiov) [inline lumimqtt/sensors.py:60]: Not sure, that it's the right behavior.

**Comment by devbis** [comment]
Did you try native MPD integration in HA to play external streams and TTS? I use HA on another device too and use MPD that runs on Lumi gateway too.

I looked at your sample projects and It looks very like the one I imagined. It requires extra static audio files included in the package. So there is no nice and clean solution with auto-adding entities to HA.

**Pull request merged**